app/controllers/orders_controller.rb

```diff
@@ -23,16 +25,27 @@ def edit
 
   def show
     @order = Order.find(params[:id])
+    if (@order.user != current_user) && !current_admin
+      render file: '/public/404'
```

Comment:
I typically prefer to use unless statements when confirming inequality. This works fine though
e.g.
 unless @order.user == current_user && current_admin